### DISCUSSION TITLE
SDK-JS: Allow to delete one or more trace

### DISCUSSION THREAD

**Discussion by Lunik** [langfuse/langfuse#3285] [Ideas]
### Describe the feature or potential improvement

Hi,

We want to run a purge script that run periodically to cleanup traces. We have seen that it is possible to delete traces from the webui but it is not possible through the python SDK (nor the public api).

Will it be possible to implement a public API endpoint to delete one trace or delete traces in batch. Also have a method in the SDK to call that API Endpoint.

**Comment by michaelgrewal** [comment]
Yes please I would like this as well for the JS TS SDK

**Comment by maharjun** [comment]
I'd like the API to handle the deletion of not only traces but also specific observations. Also, currently, If I want to null out the input and output fields I can't do it because the update function takes takes the default value as None (upon which it does not update the value).

**Reply by marcklingen** [reply]
Thanks for sharing. Would redacting inputs and outputs before logging them to Langfuse solve for your use case or do you need to do this after the fact?

**Reply by marcklingen** [reply]
Thread on this here: https://github.com/orgs/langfuse/discussions/3402

**Reply by maharjun** [reply]
For now I've managed to work around the issue independantly.

I can think of a few specific use cases for deletion that wouldn't be easy without post-facto deletion.

1. Ability to log detailed logs temporarily and then preserve only the basic logs once the purpose has been served.
2. Create log contexts lazily ie only create a span or event if it contains any logs. I have a context-based logging framework in my code base, where each time a context is entered, the relevant span or generation is created, and log messages under this context get logged within this observation. Without the ability to delete observations, I end up with a bunch of empty observations where they've been created but there's nothing in them. If we can delete them, it should be easy to delete them post-facto on context closing if there is no internal messages.
